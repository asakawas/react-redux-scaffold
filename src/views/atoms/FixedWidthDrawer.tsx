import React from 'react';
import Drawer, { DrawerProps } from '@material-ui/core/Drawer';
import styled from 'styled-components';

/**
 * The type of props of FixedWidthDrawer.
 */
export type FixedWidthDrawerProps = Readonly<
  DrawerProps & {
    /** The width (in px) of the drawer. */
    drawerWidth: number;
  }
>;

const FixedWidthDrawer = styled(({ drawerWidth, ...props }: FixedWidthDrawerProps) => (
  // Remove FixedWidthDrawerProps to suppress a runtime warning from react-dom.
  // eslint-disable-next-line react/jsx-props-no-spreading
  <Drawer aria-label="drawer" {...props} PaperProps={{ style: { width: `${drawerWidth}px` } }} />
))`
  width: ${({ drawerWidth }: FixedWidthDrawerProps) => drawerWidth}px;
`;

export default FixedWidthDrawer;
