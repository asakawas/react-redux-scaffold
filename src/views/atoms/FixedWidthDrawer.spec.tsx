import React from 'react';
import { render } from '@testing-library/react';
import FixedWidthDrawer from './FixedWidthDrawer';

describe('FixedWidthDrawer', () => {
  test('is rendered as expected', () => {
    const { asFragment } = render(<FixedWidthDrawer drawerWidth={200} open variant="persistent" />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('has specified width in its style', () => {
    const renderResult = render(<FixedWidthDrawer drawerWidth={123} open variant="persistent" />);

    const tree = renderResult.getByLabelText('drawer');
    expect(tree).toHaveStyle('width: 123px');
  });
});
