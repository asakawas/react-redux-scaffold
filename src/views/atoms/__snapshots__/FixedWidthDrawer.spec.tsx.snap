// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FixedWidthDrawer is rendered as expected 1`] = `
<DocumentFragment>
  .c0 {
  width: 200px;
}

<div
    aria-label="drawer"
    class="MuiDrawer-root MuiDrawer-docked c0"
  >
    <div
      class="MuiPaper-root MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft MuiPaper-elevation0"
      style="width: 200px;"
    />
  </div>
</DocumentFragment>
`;
